week0003/vietthaotran/python/week0003.py

```diff
@@ -0,0 +1,69 @@
+import sys
+
+class Client (object):
+	def __init__(self, name, amount):
+		self.name = name
+		self.amount = amount
+
+	def deposit(self, amount):
+		if amount > 0:
+			self.amount += amount
+			return 'True'
+		else:
+			return 'False'
+
+	def withdraw(self, amount):
+		if 0 < amount <= self.amount:
+			self.amount -= amount
+			return 'True'
+		else:
+			return 'False'
+
+class FinTech (object):
+	def __init__(self):
+		self.clients = {}
+
+	def addClient (self, name, amount):
+		if name in self.clients:
+			return 'False'
+		else:
+			client = Client(name, amount)
+			self.clients[name] = client
+			return 'True'
+
+	def deposit(self, name, amount):
+		if name in self.clients:
+			client = self.clients[name]
+			return client.deposit(amount)
+		else:
+			return 'False'
+
+	def withdraw(self, name, amount):
+		if name in self.clients:
+			client = self.clients[name]
+			return client.withdraw(amount)
+		else:
+			return 'False'
+
+	def processRequest(self, input):
```

Comment:
You should split this method from the class and treat it as an independent function/method in another class. This will increase reusability of both the class & the function itself. The FinTech class should know nothing about STDIN/input. It should have only `single responsibility` of managing `clients` and talking to `Client` objects